examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -349,17 +352,24 @@ export class TaskList extends DataObject<{ InitialState: IBaseDocumentInitialSta
 	};
 
 	protected async initializingFirstTime(props: IBaseDocumentInitialState): Promise<void> {
-		const externalTaskListId = props?.externalTaskListId;
+		const externalTaskListId = props.externalTaskListId;
 		if (externalTaskListId === undefined) {
 			throw new Error(
 				"externalTaskListId not present in instantiation. Cannot instantiate task list",
 			);
 		}
 		this._externalTaskListId = externalTaskListId;
-		// TODO: remove console.log of externalTaskListId once it is used in upcoming PRs.
-		// Linter complains about it not being used and then tried to remove
-		// _externalTaskListId as well. This log is simpy to allow building for now.
-		console.log(this.externalTaskListId);
+		// TODO: The check below is a hack for ensuring that the container is attached
+		// before creating and registering the task list, and importing the task list data
+		// form the external server. This is pretty anti-fluid, so we need a better way to
+		// do this check.
+		if (props.containerUrl === undefined) {
+			throw new Error(
+				"containerUrl is required to register the task list for external change notifications",
+			);
+		}
+		// Need to cast as IFluidResolvedUrl as IResolvedUrl does not have all url as a parameter
+		await this.registerWithCustomerService(props.containerUrl);
```

Comment:
I don't have a specific sequence of events in mind that might cause issues if we keep the registration here, before we've created the task list's internal data structures... but just to be safe I'd move it to be the last thing we call before returning from `initializingFirstTime()`.